src/InstallWizard.js

```diff
@@ -63,31 +62,83 @@ class InstallWizard extends Component {
       model: Map(),           // immutable model
       connectionInfo: undefined, // config info for external discovery services
       deployConfig: undefined, // cloud deployment configuration
+
+      //deal with update
+      currentMenuName: this.props.menuName,
+      currentOperation: undefined,
+      pages: undefined,
+      operationProps: undefined
     };
 
 
     // Indicate which of the above state variables are passed to wizard pages and can be set by them
     this.globalStateVars = ['playbookStatus', 'model', 'connectionInfo', 'deployConfig'];
+    if(this.state.currentMenuName) {
+      let extraVars = ['currentMenuName', 'currentOperation', 'steps', 'pages', 'operationProps'];
+      this.globalStateVars = this.globalStateVars.concat(extraVars);
+    }
 
 
     // Indicate which of the state variables will be persisted to, and loaded from, the progress API
     this.persistedStateVars = [
       'currentStep', 'steps', 'playbookStatus', 'connectionInfo', 'deployConfig'
     ];
+
+    if(this.state.currentMenuName) {
+      let extraVars = ['currentMenuName', 'currentOperation', 'operationProps'];
+      this.persistedStateVars = this.persistedStateVars.concat(extraVars);
+    }
```

Comment:
I'm a little confused about what this is doing, and what the impact may be of the persistedStateVars list being variable depending on other conditions.... so we're sometimes persisting 3 extra vars and sometimes not....

Reply:
those are the sets for global and persisted variables for install and update. Update need some extra vars.